install.sh

```diff
@@ -31,36 +32,62 @@ replace_icon() {
 replace_app_icon() {
   icon="$1"
   name=$(basename "$icon" .icns)
-  app="/Applications/$name.app"
+  app="$APPS/$name.app"
   if [ -d "$app" ]; then
     echo "  * $name"
     replace_icon "$icon" "$app"
   fi
 }
 
-echo
-echo ' Thanks for installing flat.icns!'
+# backup_app_icon APP
+backup_app_icon() {
+  app="$1"
+  name=$(basename "$app" .app)
+  icon="/icons_backup/$name.icns"
+  if [ -d "$app" ]; then
+    cp -f "$app/Contents/Resources/$name.icns" "/icons_backup/"
+    echo "  * $name"
+  fi
+}
 
 # need root
 if [ $EUID -ne 0 ]; then
   echo
-  echo ' ERROR: script needs root to change app icons.'
-  echo ' FIX: run with sudo or su.'
+  echo " Whoops! flat.icns needs root permissions to install your new app icons"
+  echo " To run with root permissions do: \"sudo ./install.sh\""
   echo
   exit 1
 fi
 
 echo
-echo ' Downloading icon files'
+echo " flat.icns Installer - Created by Tina Latif, Modified by Sébastien Fulmer"
+echo " GitHub Repository: https://github.com/tinalatif/flat.icns/"
+
+echo
+echo " Step 1 of 4: Downloading latest icons..."
 curl -L --progress-bar -o $ZIP $URL
+
+echo
+echo " Step 2 of 4: Unpacking icons..."
 unzip -qq -o -j $ZIP -d $DIR
 
 echo
-echo ' Replacing app icons'
+echo " Step 3 of 4: Backing up original app icons..."
+if ! [ -d "/icons_backup" ]; then
+  mkdir "/icons_backup"
+fi
+
+for app in $APPS/*.app; do
+  backup_app_icon "$app"
+done
+
+echo
+echo " Step 4 of 4: Replacing app icons..."
 for file in $DIR/*.icns; do
   replace_app_icon "$file"
 done
 
 echo
-echo ' All done! Restart your computer.'
+echo " Thanks for installing flat.icns, we hope your enjoy all your lovely new icons!"
```

Comment:
Thanks, I've fixed it in the next commit, when you're rewriting things over and over early in the morning, your bound to make a couple stupid mistakes :P